### PR TITLE
Refactor guard decorators

### PR DESCRIPTION
**Pull request by fraxachun**

### CODE REVIEW

File: demo/api/src/menus/menus.resolver.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { MainMenuObject } from "./dto/main-menu.object";
 import { MainMenuItem } from "./entities/main-menu-item.entity";
 
 @Resolver(() => MainMenuObject)
-@PublicApi()
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply demo/api/src/menus/menus.resolver.ts:13]: Maybe for the block preview when editing the main menu?

File: packages/api/cms-api/src/user-permissions/auth/user-permissions.guard.ts [hunk 2]
```diff
@@ -19,8 +19,8 @@ export class UserPermissionsGuard implements CanActivate {
     async canActivate(context: ExecutionContext): Promise<boolean> {
         const location = `${context.getClass().name}::${context.getHandler().name}()`;
 
-        if (this.getDecorator(context, "disableGlobalGuard")) return true;
-        if (this.getDecorator(context, "publicApi")) return true;
+        if (this.getDecorator(context, "disableCometGuards")) return true;
+        if (context.getClass().name === "AuthResolver") return true;
```
[the remark this answers is not in the document]
Comment (nsams) [reply packages/api/cms-api/src/user-permissions/auth/user-permissions.guard.ts:23]: hardcoding the name doesn't make it possible to implement a similar resolver in application. Do you think we will never in need for that?

Alternative idea might be something like `@RequiredPermission(RequiredPermission.none)`?

File: .changeset/great-hotels-unite.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+"@comet/cms-api": major
+---
+
+Refactor auth-decorators
+
+-   Remove `@PublicApi()`-decorator
+-   Rename `@DisableGlobalGuard()`-decorator to `@DisableCometGuards()`
+
+When using the `@DisableCometGuards()`-decorator it's not allowed to send a `x-include-invisible-content`-header.
```
Comment (nsams) [inline .changeset/great-hotels-unite.md:10]: why that? isn't that a problem when you need to make requests to an api where parts have DisableCometGuards and other not and you want to configure the gql client to always send the header (as we currently do)?

Comment (nsams) [reply .changeset/great-hotels-unite.md:10]: > what exactly we want to achieve here.

Example: Let's say we have a products graphql query that is used in a site with a client side component for a product search
- for admin users: in site preview where it should also show invisible products (if the user activated the toggle)
- for any (also non authed) user: the header must not allowed

(A BFF would not have this problem)

**Review by nsams — APPROVED** [review]

File: packages/api/cms-api/src/public-upload/public-uploads.controller.ts [hunk 4]
```diff
@@ -2,20 +2,20 @@ import { EntityManager } from "@mikro-orm/postgresql";
...
-import { PublicApi } from "../auth/decorators/public-api.decorator";
+import { DisableCometGuards } from "../auth/decorators/disable-comet-guards.decorator";
 import { PublicUploadFileUploadInterface } from "./dto/public-upload-file-upload.interface";
 import { PublicUpload } from "./entities/public-upload.entity";
 import { PublicUploadFileInterceptor } from "./public-upload-file.interceptor";
 import { PublicUploadsService } from "./public-uploads.service";
 
 @Controller("public-upload/files")
-@PublicApi()
+@DisableCometGuards()
```
Comment (johnnyomair) [inline packages/api/cms-api/src/public-upload/public-uploads.controller.ts:12]: We should remove it here and as it's already set in the upload function. Also, setting the decorator on controllers/decorators could lead to unintentionally creating open routes

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**